File: LeetCode/Easy/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_dict = {}
        
        for i in range(len(nums)):
            value = target - nums[i]
            
            if value in index_dict:
                return [index_dict[value], i]
            
            index_dict[nums[i]] = i

File: LeetCode/Easy/0001-two-sum/0001-two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int value = target - nums[i];
            
            if (indexMap.containsKey(value)) {
                return new int[] { indexMap.get(value), i };
            }
            
            indexMap.put(nums[i], i);
        }

        throw new IllegalArgumentException();
    }
}